pyuvsim/telescope.py

```diff
@@ -37,3 +41,278 @@ def __eq__(self, other):
+                    for key in mismatches]
+                )
+                raise ValueError('UVBeam parameters do not match '
+                                 'those currently set: \n\t' + compare)
+
+            self._str_beam_list[ind] = self._obj_to_str(value)
+        else:
+            value = self._str_to_obj(value)
+            self._obj_beam_list[ind] = value
+            self._scrape_uvb_params(self._obj_beam_list, strict=False)
+
+    def append(self, value):
+        """
+        Append to the beam list, converting objects to strings if in object mode,
+        or vice versa if in string mode.
+
+        If in string mode, this will overwrite the uvb_params with the added
+        object's parameters.
+        """
+        if self.string_mode:
+            self._str_beam_list.append('')
+        else:
+            self._obj_beam_list.append('')
+        try:
+            self.__setitem__(-1, value)
+        except ValueError as err:
+            if self.string_mode:
+                self._str_beam_list.pop()
+            else:
+                self._obj_beam_list.pop()
+            raise err
+
+    def _str_to_obj(self, beam_model):
+        # Convert beam strings to objects.
+        if isinstance(beam_model, (AnalyticBeam, UVBeam)):
+            return beam_model
+        if beam_model.startswith('analytic'):
+            bspl = beam_model.split('_')
+            model = bspl[1]
+
+            to_set = {}
+            for extra in bspl[2:]:
+                par, val = extra.split('=')
+                full = self._float_params[par]
+                to_set[full] = float(val)
+
+            return AnalyticBeam(model, **to_set)
+
+        path = beam_model  # beam_model = path to beamfits
+        uvb = UVBeam()
+
+        uvb.read_beamfits(path)
+        for key, val in self.uvb_params.items():
+            setattr(uvb, key, val)
+        uvb.extra_keywords['beam_path'] = path
+        return uvb
+
+    def _obj_to_str(self, beam_model):
+        # Convert beam objects to strings that may generate them.
+
+        if isinstance(beam_model, str):
+            return beam_model
+        if isinstance(beam_model, AnalyticBeam):
+            btype = beam_model.type
+
+            bm_str = 'analytic_' + btype
+            for abbrv, full in self._float_params.items():
+                val = getattr(beam_model, full)
+                if val is not None:
+                    bm_str += '_' + abbrv + '=' + str(val)
+            return bm_str
+
+        # If not AnalyticBeam, it's UVBeam.
+        try:
+            path = beam_model.extra_keywords['beam_path']
+        except KeyError:
+            raise ValueError("Need to set 'beam_path' key in extra_keywords for UVBeam objects.")
+
+        return path
+
+    def set_str_mode(self):
+        """
+        Set string_mode = True, converting object beams to their
+        string representations and putting UVBeam parameters into the
+        uvb_params dictionary.
```

Comment:
This docstring will error with flake8 when we turn on docstring testing in Steven's PR -- The first line should be a single line sentence (ending in a period). More details should follow in text after a blank line.